src/dashboard/apps/renewable/forms.py

```diff
@@ -0,0 +1,184 @@
+
+from apps.renewable.models import Renewable
+
+
+class RenewableCheckboxInput(forms.CheckboxInput):
+    """Custom CheckboxInput widget for rendering a checkbox input field."""
+
+    template_name = "renewable/forms/widgets/checkbox.html"
+
+
+class RenewableForm(forms.ModelForm):
+    """Form for managing renewable meter readings."""
+
+    BASE_WIDGET_ATTRS = {
+        "class": "fr-input",
+    }
+
+    meter_reading = forms.FloatField(
+        required=False,
+        widget=forms.NumberInput(
+            attrs={
+                **BASE_WIDGET_ATTRS,
+                "type": "number",
+                "min": "0",
+                "step": "0.01",
+            }
+        ),
+    )
+    collected_at = forms.DateField(
+        required=False,
+        widget=forms.DateInput(
+            attrs={
+                **BASE_WIDGET_ATTRS,
+                "type": "date",
+            }
+        ),
+    )
+
+    class Meta:  # noqa: D106
+        model = Renewable
+        fields = ["meter_reading", "collected_at"]
+
+    def __init__(self, *args, **kwargs):
+        """Initializes the form with error classes and aria-describedby attributes."""
+        super().__init__(*args, **kwargs)
+        self._add_error_classes_to_fields()
+
+    def _add_error_classes_to_fields(self):
+        """Add error widget attributes to fields with errors."""
+        if not self.errors:
+            return
+
+        for field_name, field in self.fields.items():
+            if field_name in self.errors:
+                current_class = field.widget.attrs.get("class", "")
+
+                field.widget.attrs.update(
+                    {
+                        "class": f"{current_class} fr-input--error",
+                        "aria-describedby": f"{ field_name }-desc-error",
+                    }
+                )
+
+    def clean_collected_at(self):
+        """Validates the collected_at field."""
+        collected_at = self.cleaned_data.get("collected_at")
+        meter_reading = self.cleaned_data.get("meter_reading")
+
+        if meter_reading and not collected_at:
+            raise forms.ValidationError(
+                _("Collected date is required if a meter reading is provided")
+            )
+
+        if not collected_at:
+            return collected_at
+
+        # TODO: This logic should be moved to the models or a helper function.
+        # TODO: Review the logic: readings meters should be taken between the last day
+        #  of the previous quarter and 'x' days before the end of the previous quarter.
+        # TODO: the 'x' days should be configurable via the settings.
+        today = datetime.datetime.now().date()
+        min_date = today - datetime.timedelta(days=10)
+
+        if collected_at > today:
+            raise forms.ValidationError(_("The date cannot be in the future"))
+        elif collected_at < min_date:
+            raise forms.ValidationError(_("The date cannot be earlier than 10 days"))
```

Comment:
In case rules change:

```suggestion
        elif collected_at < min_date:
            raise forms.ValidationError(_(f"The date cannot be earlier than {min_date}"))
```